### DISCUSSION TITLE
Is 'mip' no longer supported?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16286] [General]
From the following documentation...
https://docs.micropython.org/en/latest/reference/packages.html

>>> import mip
>>> mip.install("pkgname")  # Installs the latest version of "pkgname" (and dependencies)

But I get the following...
```
Installing umqttsimple (latest) from https://micropython.org/pi/v2 to /lib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mip/__init__.py", line 1, in install
  File "mip/__init__.py", line 1, in _install_package
  File "mip/__init__.py", line 1, in _install_json
  File "requests/__init__.py", line 201, in get
  File "requests/__init__.py", line 79, in request
OSError: -202
```

Just trying to add MQTT support to my MicroPython project, i.e. just publishing to MQTT.

**Comment by peterhinch** [comment]
This works here
```py
Connected to MicroPython at /dev/ttyACM0
Use Ctrl-] or Ctrl-x to exit this shell

>>> import mip
>>> mip.install("umqtt.simple")
Installing umqtt.simple (latest) from https://micropython.org/pi/v2 to /lib
Copying: /lib/umqtt/simple.mpy
Done
>>> 
```

**Comment by andypiper** [comment]
Is your board already connected to a network? It looks like `requests` is failing.

**Comment by bidrohini** [comment]
The error you’re seeing can be due to a network failure when trying to download the package. Try to verify the internet connection by accessing a website or running any other network-related code.
For example, you can try using a simple HTTP request:
import socket
socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect(("google.com", 80))

**Comment by Jibun-no-Kage** [comment]
Initial issue was the ESP32 module had indeed drop off the network.  So attempted again...
```
MicroPython v1.24.0 on 2024-10-25; Generic ESP32 module with ESP32

Type "help()" for more information.

>>> import network
>>> theNetwork = network.WLAN(network.STA_IF)
>>> theNetwork.active(True)
True
>>> theNetwork.connect('****', '*****')
>>> theNetwork.isconnected()
True
>>> import mip
>>> mip.install("umqtt.simple")
Installing umqtt.simple (latest) from https://micropython.org/pi/v2 to /lib
```
Now the odd thing is... the installing sequence has been stuck there for about 15 minutes.  Appears to have hung?

If I try the URL directly... I get error 403.  But not sure that not expected with any direct access attempt to the URL?

**Reply by scruss** [reply]
Is your router normally flaky? Because I had `umqtt.simple` install in  couple of seconds on a 1 MB ESP01s:
```
MPY: soft reboot
Starting network ...
network config: ('192.168.2.124', '255.255.255.0', '192.168.2.1', '192.168.2.1')
hostname      : esp01a
MicroPython v1.24.0 on 2024-10-25; ESP module (1M) with ESP8266
Type "help()" for more information.
>>> import mip
>>> mip.install("umqtt.simple")
Installing umqtt.simple (latest) from https://micropython.org/pi/v2 to /lib
Copying: /lib/umqtt/simple.mpy
Done
```
If you have a direct connection to the board, you can use `mpremote mip install ...` rather than the REPL. It allows you to use mip on boards with no wifi connection.

Also, I find [mim](https://mim.oliverrobson.tech/) extremely useful for finding mip package names.

**Comment by Jibun-no-Kage** [comment]
Oh... assuming the following code would generate an error if no connection of socket?
```
>>> import socket
>>> socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect(("google.com", 80))
>>> 
```
And... Showing current status...
```
>>> theNetwork
<WLAN>
>>> theNetwork.ifconfig()
('192.168.1.216', '255.255.255.0', '192.168.1.3', '192.168.1.253')
>>> 
```
